src/Filter/FilterCollection.php

```diff
@@ -62,8 +65,12 @@ public function input($file, $content)
      */
     public function output($target, $content)
     {
+        $shouldApplyFilterGeneral = (!$this->debug || php_sapi_name() === 'cli');
         foreach ($this->filters as $filter) {
-            $content = $filter->output($target, $content);
+            $settings = $filter->settings();
+            if ((array_key_exists('allowDebugMode', $settings) && !!$settings['allowDebugMode']) || $shouldApplyFilterGeneral) {
```

Comment:
`PostCSS' autoprefixer` can be applied in both debug and production mode. Others such as `uglifyjs` should be definitely applied in production mode only.